AppKit/CPText.j

```diff
@@ -46,8 +76,283 @@ CPWritingDirectionLeftToRight   = 0;
 CPWritingDirectionRightToLeft   = 1;
 
 @typedef CPTextAlignment
-CPLeftTextAlignment      = 0;
-CPRightTextAlignment     = 1;
-CPCenterTextAlignment    = 2;
-CPJustifiedTextAlignment = 3;
-CPNaturalTextAlignment   = 4;
\ No newline at end of file
+CPLeftTextAlignment             = 0;
+CPRightTextAlignment            = 1;
+CPCenterTextAlignment           = 2;
+CPJustifiedTextAlignment        = 3;
+CPNaturalTextAlignment          = 4;
+
+/*
+    CPText notifications
+*/
+CPTextDidBeginEditingNotification = @"CPTextDidBeginEditingNotification";
+CPTextDidChangeNotification = @"CPTextDidChangeNotification";
+CPTextDidEndEditingNotification = @"CPTextDidEndEditingNotification";
+
+/*
+    CPTextView Notifications
+*/
+CPTextViewDidChangeSelectionNotification        = @"CPTextViewDidChangeSelectionNotification";
+CPTextViewDidChangeTypingAttributesNotification = @"CPTextViewDidChangeTypingAttributesNotification";
+
+/*
+    FIXME: move these to CPAttributed string
+    Make use of attributed keys in AppKit
+*/
+CPFontAttributeName = @"CPFontAttributeName";
+CPForegroundColorAttributeName = @"CPForegroundColorAttributeName";
+CPBackgroundColorAttributeName = @"CPBackgroundColorAttributeName";
+CPShadowAttributeName = @"CPShadowAttributeName";
+CPUnderlineStyleAttributeName = @"CPUnderlineStyleAttributeName";
+CPSuperscriptAttributeName = @"CPSuperscriptAttributeName";
+CPBaselineOffsetAttributeName = @"CPBaselineOffsetAttributeName";
+CPAttachmentAttributeName = @"CPAttachmentAttributeName";
+CPLigatureAttributeName = @"CPLigatureAttributeName";
+CPKernAttributeName = @"CPKernAttributeName";
+
+@implementation CPText : CPView
+{
+    BOOL       _isEditable       @accessors(getter=isEditable, setter=setEditable:);
+    BOOL       _isSelectable     @accessors(getter=isSelectable, setter=setSelectable:);
+    BOOL       _isRichText       @accessors(getter=isRichText, setter=setRichText:);
+}
+
+- (void)setSelectable:(BOOL)flag
+{
+    _isSelectable = flag;
+
+    if (!flag)
+        [self setEditable:flag];
+}
+
+- (void)setEditable:(BOOL)flag
+{
+    _isEditable = flag;
```

Comment:
We can bind this ivar. We need to add the willChangeValueForKey and didChangeValueForKey